compiler/tests/mod.rs

```diff
@@ -181,12 +181,23 @@ pub(crate) fn expect_compiler_error(program: EdwardsTestCompiler) -> CompilerErr
     program.generate_constraints_helper(&mut cs).unwrap_err()
 }
 
-// pub(crate) fn expect_synthesis_error(program: EdwardsTestCompiler) {
-//     let mut cs = TestConstraintSystem::<Fq>::new();
-//     let _output = program.generate_constraints_helper(&mut cs).unwrap();
-//
-//     assert!(!cs.is_satisfied());
-// }
+pub(crate) fn expect_dynamic_check_error(error: CompilerError) {
+    let is_dynamic_check = match error {
+        CompilerError::DynamicCheckError(_) => true,
+        _ => false,
+    };
+
+    assert!(is_dynamic_check)
+}
+
+pub(crate) fn expect_static_check_error(error: CompilerError) {
+    let is_static_check = match error {
+        CompilerError::StaticCheckError(_) => true,
+        _ => false,
+    };
+
+    assert!(is_static_check)
+}
```

Comment:
these 2 functions could use the `matches!` macro for brevity; both could become single-liners with it